pkg/network/dns/types.go

```diff
@@ -32,10 +33,17 @@ const (
 	TypeURI   QueryType = 256 // URI RR [RFC7553]
 )
 
+// *stringintern.Value gives a clearer name than "value"
+//type *stringintern.Value *stringintern.Value
+
+// MapByKeyNameType provides a type name for the map of
+// DNS stats based on the host key->the lookup name->querytype
+type MapByKeyNameType map[Key]map[*stringintern.Value]map[QueryType]Stats
```

Comment:
What do you think about `MapByKeyByNameByType`? I think that makes it clear that `Key`, `Name`, and `Type` are separate key types, rather than a single key type called `KeyNameType`.